execution/executor/src/lib.rs

```diff
@@ -544,7 +546,11 @@ where
         ensure!(
             txns_to_retry.is_empty(),
             "The transaction at version {} got the status of 'Retry'",
-            num_txns - txns_to_retry.len() + first_version as usize,
+            num_txns
+                .checked_sub(txns_to_retry.len())
+                .expect("Integer overflow occurred")
+                .checked_add(first_version as usize)
+                .expect("Integer overflow occurred"),
```

Comment:
do we really want to crash here? Your lint won't be happy since you're in a function that returns a result :P

Reply:
can we maybe avoid the checked stuff by modifying the log message?

We could log num_txns, txns_to_retry, and first_version and this would avoid doing math on them

Reply:
fixed :D